utils/s3/client.hh

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <seastar/core/sstring.hh>
+#include <seastar/core/shared_ptr.hh>
+#include <seastar/http/client.hh>
+
+using namespace seastar;
+
+namespace s3 {
+
+struct range {
+    uint64_t off;
+    size_t len;
+};
+
+class client : public enable_shared_from_this<client> {
+    sstring _host;
+    http::experimental::client _http;
+
+    struct private_tag {};
+    future<std::vector<temporary_buffer<char>>> get_object(sstring object_name, std::optional<std::pair<uint64_t, size_t>> range);
+
+public:
+    explicit client(ipv4_addr addr, private_tag);
+    static shared_ptr<client> make(ipv4_addr addr);
+
+    future<uint64_t> get_object_size(sstring object_name);
+    future<temporary_buffer<char>> get_object_contiguous(sstring object_name, std::optional<range> range = {});
```

Comment:
sounds good we'll have only one interface for retrieving a range of an object into a contiguous buffer. sstable layer must be responsible anyway for not reading a large range in a single iteration.